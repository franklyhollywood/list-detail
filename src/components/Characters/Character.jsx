import { Link } from 'react-router-dom';
import { Center, Button } from '@chakra-ui/react';

export default function Character({ character }) {
  return (
    <>
      <Center>
        <img
          src={character.photoUrl}
          alt={character.name}
          width="500"
          height="600"
        />
      </Center>
      <Center>
        <div>name: {character.name} </div>
        <div>position: {character.position}</div>
        <div>allies: {character.allies}</div>
        <p>
          <Link to="/">
            <Button>BACK</Button>
          </Link>
        </p>
      </Center>
    </>
  );
}
