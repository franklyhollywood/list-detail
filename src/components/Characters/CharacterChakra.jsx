import { Box, Badge, Image, Button } from '@chakra-ui/react';
import { Link } from 'react-router-dom';

export default function CharacterChakra({ character }) {
  return (
    <Box
      maxW="sm"
      minW="sm"
      borderWidth="10px"
      borderRadius="lg"
      overflow="hidden"
      fit="cover"
    >
      <Image
        minW="sm"
        align="top center"
        src={character.photoUrl}
        alt={character.name}
      />

      <Box p="6">
        <Box display="flex" alignItems="baseline">
          <Badge borderRadius="full" px="2" colorScheme="teal">
            Allies: {character.allies}
          </Badge>
          <Box
            color="gray.500"
            fontWeight="semibold"
            letterSpacing="wide"
            fontSize="xs"
            textTransform="uppercase"
            ml="2"
          >
            Enemies: {character.enemies}
          </Box>
        </Box>
        <Button size="xs" margin="2" colorScheme="blue">
          <Link to={`/character/${character._id}`}>
            <Box
              mt="1"
              fontWeight="semibold"
              as="h4"
              lineHeight="tight"
              isTruncated
            >
              Details: {character.name}
            </Box>
          </Link>
        </Button>
        <Box>
          {character.affiliation}
          <Box as="span" color="gray.600" fontSize="sm"></Box>
        </Box>
      </Box>
    </Box>
  );
}
