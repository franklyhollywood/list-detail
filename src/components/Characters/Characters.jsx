import { Link } from 'react-router-dom';

export default function Characters({ characters }) {
  return (
    <ul>
      {characters.map((character) => (
        <li key={character._id}>
          <Link to={`/character/${character._id}`}>
            <img src={character.photoUrl} alt={character.name} width="200" />
            <div>name: {character.name}</div>
          </Link>
        </li>
      ))}
    </ul>
  );
}
