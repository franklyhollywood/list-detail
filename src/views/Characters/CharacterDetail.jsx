import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { getCharacter } from '../../services/avatarAPI.js';
import Character from '../../components/Characters/Character.jsx';

export default function CharacterDetail() {
  const [character, setCharacter] = useState({});
  const [loading, setLoading] = useState(true);
  const { id } = useParams();

  useEffect(() => {
    const loadNewCharacter = async () => {
      const returnedCharacter = await getCharacter(id);
      setCharacter(returnedCharacter);
      setLoading(false);
    };
    loadNewCharacter();
  }, [id]);

  return (
    <>
      <div>
        {loading ? (
          <h1>loading character...</h1>
        ) : (
          <Character character={character} />
        )}
      </div>
    </>
  );
}
