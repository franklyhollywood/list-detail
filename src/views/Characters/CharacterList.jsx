import { getCharacters } from '../../services/avatarAPI.js';
import { useState, useEffect } from 'react';
import Characters from '../../components/Characters/Characters.jsx';

export default function CharacterList() {
  const [characters, setCharacters] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const loadAllCharacters = async () => {
      const returnedCharacters = await getCharacters();
      setCharacters(returnedCharacters);
      setLoading(false);
    };
    loadAllCharacters();
  }, []);

  return (
    <>
      {loading ? <h2>Loading...</h2> : <Characters characters={characters} />}
    </>
  );
}
