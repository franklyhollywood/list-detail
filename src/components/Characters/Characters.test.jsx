import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Characters from './Characters.jsx';

it('renders the characters view', () => {
  const characters = [
    {
      _id: '5cf5679a915ecad153ab68c8',
      allies: 'Royal Earthbender Guards',
      enemies: 'Chin',
      photoUrl:
        'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
    },
  ];

  const { container } = render(
    <MemoryRouter>
      <Characters characters={characters} />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
