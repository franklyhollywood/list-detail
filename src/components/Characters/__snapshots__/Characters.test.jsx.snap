// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the characters view 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/character/5cf5679a915ecad153ab68c8"
      >
        <img
          src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
          width="200"
        />
        <div>
          name: 
        </div>
      </a>
    </li>
  </ul>
</div>
`;
